monitoring/voltage-monitoring/src/test/java/com/farao_community/farao/monitoring/voltage_monitoring/VoltageMonitoringTest.java

```diff
@@ -346,4 +347,159 @@ void testMultipleVoltageValuesPerVoltageLevel() {
             "Network element VL46 at state preventive has a voltage of 143 - 148 kV."),
             voltageMonitoringResult.printConstraints());
     }
+
+    public void setUpCracFactory(String networkFileName) {
+        network = Network.read(networkFileName, getClass().getResourceAsStream("/" + networkFileName));
+        crac = CracFactory.findDefault().create("test-crac");
+    }
+
+    public void mockPreventiveState() {
+        vcPrev =  addVoltageCnec("vcPrev", Instant.PREVENTIVE, null, "VL45", 145., 150.);
+        crac.newNetworkAction()
+                .withId("Open L1 - 1")
+                .newTopologicalAction().withNetworkElement("L1").withActionType(ActionType.OPEN).add()
+                .newOnVoltageConstraintUsageRule().withInstant(Instant.PREVENTIVE).withVoltageCnec(vcPrev.getId()).add()
+                .add();
+    }
+
+    @Test
+    void testDivergentLoadFlowDuringInitialLoadFlow() {
+        setUpCracFactory("networkKO.xiidm");
+        mockPreventiveState();
+
+        runVoltageMonitoring();
+
+        assertEquals(UNKNOWN, voltageMonitoringResult.getStatus());
+        voltageMonitoringResult.getAppliedRas().forEach((state, networkActions) -> assertTrue(networkActions.isEmpty()));
+        assertEquals(Double.NaN, voltageMonitoringResult.getMinVoltage(vcPrev));
+        assertEquals(Double.NaN, voltageMonitoringResult.getMaxVoltage(vcPrev));
+    }
+
+    @Test
+    void testDivergentLoadFlowAfterApplicationOfRemedialAction() {
+        setUpCracFactory("network2.xiidm");
+
+        crac.newContingency().withId("co").withNetworkElement("L1").add();
+        VoltageCnec vc =  addVoltageCnec("vc", Instant.CURATIVE, "co", "VL1", 390., 399.);
+
+        crac.newNetworkAction()
+                .withId("Open L2 - 1")
+                .newTopologicalAction().withNetworkElement("L2").withActionType(ActionType.OPEN).add()
+                .newOnVoltageConstraintUsageRule().withInstant(Instant.CURATIVE).withVoltageCnec(vc.getId()).add()
+                .add();
+
+        runVoltageMonitoring();
+
+        assertEquals(HIGH_VOLTAGE_CONSTRAINT, voltageMonitoringResult.getStatus());
+        assertEquals(1, voltageMonitoringResult.getAppliedRas().size());
```

Comment:
this doesn't test that the correct RA is listed in voltageMonitoringResult.getAppliedRas(), does it?

Reply:
This works, but it is very complicated to understand and the intention is not clear. How about this instead (you can't commit the change because I couldn't select the right lines to modify):
```suggestion
                RemedialAction <?> networkAction = crac.newNetworkAction()
                .withId("Open L2 - 1")
                .newTopologicalAction().withNetworkElement("L2").withActionType(ActionType.OPEN).add()
                .newOnVoltageConstraintUsageRule().withInstant(Instant.CURATIVE).withVoltageCnec(vc.getId()).add()
                .add();

        runVoltageMonitoring();

        assertEquals(HIGH_VOLTAGE_CONSTRAINT, voltageMonitoringResult.getStatus());
        assertEquals(1, voltageMonitoringResult.getAppliedRas().size());
        assertEquals(Set.of(networkAction), voltageMonitoringResult.getAppliedRas().get(crac.getState("co", Instant.CURATIVE)));
```

Reply:
Fixed this case